src/bump/callHook.ts

```diff
@@ -16,6 +16,8 @@ export async function callHook(
 
   for (const pkg of affectedPackages) {
     const packageInfo = packageInfos[pkg];
-    await hook(path.dirname(packageInfo.packageJsonPath), packageInfo.name, packageInfo.version);
+    const packagePath = path.dirname(packageInfo.packageJsonPath);
+
+    await hook(packagePath, packageInfo.name, packageInfo.version, packageInfos);
```

Comment:
@ecraig12345 Should we clone package infos before passing? We're susceptible to side effects by hooks in the current calling pattern

Reply:
Marked param as Readonly<> in the typing of hooks. Doesn't technically stop a consumer from casting as writable and causing side effects. lmk what you think